starport-blog/04-http-create-post.md

```diff
@@ -0,0 +1,282 @@
+    const account = data.result.value;
+    const chain_id = "blog";
+    const meta = {
+      // Making sure both sequence and account_number are strings
+      // Sequence number changes every time we submit a new transaction
+      sequence: `${account.sequence}`,
+      // Account number stays the same
+      account_number: `${account.account_number}`,
+      chain_id,
+    };
+    const req = {
+      base_req: {
+        chain_id,
+        from: wallet.address,
+      },
+      creator: wallet.address,
+      title,
+    };
+    // Fetching am unsigned transaction
+    axios.post(`${API}/blog/posts`, req).then(({ data }) => {
+      const tx = data.value;
+      // Signing the transaction with the private key and meta info
+      const stdTx = signTx(tx, meta, wallet);
+      // Preparing transaction for broadcasting
+      // "block" will make sure we get the response after
+      // the transaction is included in the block
+      const txBroadcast = createBroadcastTx(stdTx, "block");
+      const params = {
+        headers: {
+          "Content-Type": "application/json",
+        },
+      };
+      // Sending our post to be processed by the server
+      axios.post(`${API}/txs`, txBroadcast, params).then(() => {
+        // Fetch a new list of posts after we've successfully
+        // created a new post
+        fetchPosts();
+      });
+    });
+  });
+};
+```
+
+Run init to initialize our app:
+
+```js
+init();
+```
+
+## Creating an account
+
+Before we start using our application we need to make sure that the account we have generated exists on our chain. To do so we will send a nominal amount of tokens from an existing account to a new one.
+
+```sh
+blogcli keys show user1
+```
+
+You will get information about one of the existing accounts (your values will be different):
+
+```json
+{
+  "name": "user1",
+  "type": "local",
+  "address": "cosmos1wt47yve6l29yjtxtsajhltr2vqhf7mpw5n6fx6",
+  "pubkey": "cosmospub1addwnpepq03v7d6q4yt4nalj74elq8l5498wd9krcx92mxudkarj8aapy0qjvfaga8z"
+}
+```
+
+Transfer some tokens from this account to the new one:
+
+```sh
+blogcli tx send cosmos1wt47yve6l29yjtxtsajhltr2vqhf7mpw5n6fx6 cosmos152gzu3vzf7g9tu46vszgpac24lwr48vc8k8kkh 10foo --from=user1
+```
+
+Notice that the sender address `cosmos1wt47yve6l29yjtxtsajhltr2vqhf7mpw5n6fx6` is the one we got from running `blogcli keys show me` and the receiver account address `cosmos152gzu3vzf7g9tu46vszgpac24lwr48vc8k8kkh` is the one we have generated from the mnemonic in browser.
+
+In this guide we're activating accounts manually, but in production apps you might want to do it as part of a signing up process.
+
+Now that you that we've created the app and set up our account, let's run it!
+
+```
+npm run serve
+```
+
+Open `http://localhost:1234/` and try creating new posts. New posts should show up in the list after 2-4 seconds after submission. Notice that posts persist after you refresh the page.
```

Comment:
I get this error in the browser console:
```
xhr.js:178 POST http://localhost:8010/proxy/blog/posts 405 (Method Not Allowed)
```

Reply:
Seems like proxy didn't work. I've created a repo with branches for each step. Checked multiple times, couldn't reproduce this error.

https://github.com/fadeev/tutorial-blog/tree/04-http-create-posts

Reply:
There was one more error: `foo` instead of `token`. Starport used to init with `foo` (like Scavenge), but we have replaced it with `token`.